### PR TITLE
feat(build): support the first build

### PR DESCRIPTION
**Pull request by oliviertassinari**
Start working on #232 as well as #188 also improving the fallbackToMaster logic and the `/builds` response time.

Missing:
- [x] Testing the fallbackToMaster of `baseCompare`
- [x] Testing `build`
- [x] Local recette
capture d ecran 2017-03-31 a 23 51 06

Target: https://github.com/argos-ci/argos/milestone/4 🎉

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ yarn dev:review # run webpack
 #### Create a migration
 
 ```sh
-knex migrate:make my_migration
+./node_modules/.bin/knex migrate:make my_migration
```
Comment (neoziro) [inline README.md]: Add `PATH=$PATH:./node_modules/.bin` to your .profile

File: src/modules/baseCompare/baseCompare.js [hunk 2]
```diff
@@ -35,62 +39,54 @@ async function getBaseScreenshotBucket({
...
-  const user = await repository.getUsers().limit(1).first()
+  build = await build.$query().eager('[repository, compareScreenshotBucket]')
+  const user = await build.repository.getUsers().limit(1).first()
 
   if (!user) {
     return {
-      user: null,
-      compareCommitFound: false,
-      baseScreenshotBucket: null,
+      baseScreenshotBucket: await fallbackToMaster(build),
```
Comment (neoziro) [inline src/modules/baseCompare/baseCompare.js]: Bizarre de retourner un objet vu le nom de la méthode O_o

File: src/modules/build/createBuildDiffs.js [hunk 3]
```diff
@@ -1,29 +1,52 @@
...
+    build,
+  })
+
+  return transaction(ScreenshotDiff, Build, async (ScreenshotDiff, Build) => {
+    if (build.baseScreenshotBucket !== baseScreenshotBucket) {
+      build.baseScreenshotBucket = await baseScreenshotBucket.$query()
+        .eager('screenshots')
+
+      await Build.query()
+        .patch({
```
Comment (neoziro) [inline src/modules/build/createBuildDiffs.js]: One line if < 3 args and max-len < 80

File: src/modules/build/createBuildDiffs.js [hunk 4]
```diff
@@ -1,29 +1,52 @@
...
+  const {
+    baseScreenshotBucket,
+  } = await baseCompare({
+    baseCommit: 'master',
+    compareCommit: build.compareScreenshotBucket.commit,
+    build,
+  })
+
+  return transaction(ScreenshotDiff, Build, async (ScreenshotDiff, Build) => {
+    if (build.baseScreenshotBucket !== baseScreenshotBucket) {
```
Comment (neoziro) [inline src/modules/build/createBuildDiffs.js]: On ne devrait pas mieux comparer des ids ?

File: src/modules/baseCompare/baseCompare.js (hunk 2 above)
Comment (neoziro) [inline src/modules/baseCompare/baseCompare.js]: T'as dû itérer et oublier de changer non ?

File: src/modules/build/createBuildDiffs.js (hunk 3 above)
Comment (oliviertassinari) [inline src/modules/build/createBuildDiffs.js]: We should use prettier

File: src/modules/build/createBuildDiffs.js (hunk 4 above)
Comment (oliviertassinari) [inline src/modules/build/createBuildDiffs.js]: Je vais supprimer cette ligne, c'est une optimisation.

File: src/modules/baseCompare/baseCompare.js (hunk 2 above)
Comment (oliviertassinari) [inline src/modules/baseCompare/baseCompare.js]: I'm gonna make that flat.

File: src/server/models/ScreenshotDiff.js [hunk 5]
```diff
@@ -18,7 +18,7 @@ export default class ScreenshotDiff extends BaseModel {
       ],
       properties: {
         buildId: { type: 'string' },
-        baseScreenshotId: { type: 'string' },
+        baseScreenshotId: { type: ['string', null] },
         compareScreenshotId: { type: 'string' },
```
Comment (oliviertassinari) [inline src/server/models/ScreenshotDiff.js]: We could imagine the same change when a screenshot is removed or renamed.

**Pull request merged**